controller/konnect/ops/ops_controlplane_test.go

```diff
@@ -448,3 +457,56 @@ func TestUpdateControlPlane(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateAndUpdateControlPlane_KubernetesMetadataConsistency(t *testing.T) {
+	var (
+		ctx = context.Background()
+		cp  = &konnectv1alpha1.KonnectGatewayControlPlane{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "KonnectGatewayControlPlane",
+				APIVersion: "konnect.konghq.com/v1alpha1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "cp-1",
+				Namespace: "default",
+				UID:       k8stypes.UID(uuid.NewString()),
+			},
+		}
+		sdk = &MockControlPlaneSDK{}
+	)
+
+	t.Log("Triggering CreateControlPlane and capturing generated tags")
+	sdk.EXPECT().
+		CreateControlPlane(ctx, mock.Anything).
+		Return(&sdkkonnectops.CreateControlPlaneResponse{
+			ControlPlane: &sdkkonnectcomp.ControlPlane{
+				ID: "12345",
+			},
+		}, nil)
+	err := createControlPlane(ctx, sdk, cp)
+	require.NoError(t, err)
+	require.Len(t, sdk.Calls, 1)
+	call := sdk.Calls[0]
+	require.Equal(t, "CreateControlPlane", call.Method)
+	require.IsType(t, sdkkonnectcomp.CreateControlPlaneRequest{}, call.Arguments[1])
+	capturedCreateLabels := call.Arguments[1].(sdkkonnectcomp.CreateControlPlaneRequest).Labels
```

Comment:
Can't we provide the expected labels to the `sdk.EXPECT().CreateControlPlane(...)` call above instead 🤔 ?

Similarly to how it's done in https://github.com/Kong/gateway-operator/blob/a3b231ea17596910065f2964f00c20102afe41e7/controller/konnect/ops/ops_kongservice_test.go#L383-L398

Reply:
Yeah, my initial intention was to focus just on the labels in the test and ensure they're the same between create and update calls. The mocking framework doesn't allow matching only a single field in an expected argument so I tried extracting this which turned out quite verbatim as you noticed. I changed the test to explicitly expect labels using the mock expectations which I believe joins these two worlds. 

I also changed the way I test the rest of the entities' tags output - focused on testing the mapping functions only (`kongConsumerToSDKConsumerInput`, etc.) rather than the whole `Create` function, to ensure they all return the expected tags.